### PR TITLE
Adds an admin-only version of the chef's hat that allows mice inside of the hat to have a 100% chance to relay their movements to the person wearing the hat

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

See title.

## Why It's Good For The Game

https://www.youtube.com/watch?v=YZvbCgnCJoA

## Changelog
:cl: ATHATH
add: A secret cabal of mice have definitely not been controlling people wearing special admin-only chef's hats. Move along, citizen.
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/jobs.dm [hunk 1]
```diff
@@ -26,7 +32,7 @@
 	return(FIRELOSS)
 
 /obj/item/clothing/head/chefhat/relaymove(mob/user, direction)
-	if(!istype(user, /mob/living/simple_animal/mouse) || !isliving(loc) || !prob(20))
+	if(!istype(user, /mob/living/simple_animal/mouse) || !isliving(loc) || !prob(mousecontrolprobability))
 		return
 	var/mob/living/L = loc
 	step_towards(L, get_step(L, direction))
```
Comment (ShizCalev) [inline code/modules/clothing/head/jobs.dm:38]: might need a incapacitated check here for L, unless step already handles that

File: code/modules/clothing/head/jobs.dm [hunk 2]
```diff
@@ -15,6 +15,12 @@
 	dynamic_hair_suffix = ""
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/chefhat
 	dog_fashion = /datum/dog_fashion/head/chef
+	///the chance that the movements of a mouse inside of this hat get relayed to the human wearing the hat
+	var/mousecontrolprobability = 20
+
+/obj/item/clothing/head/chefhat/iamassumingdirectcontrol
```
Comment (ShizCalev) [inline code/modules/clothing/head/jobs.dm:21]: snake case

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/jobs.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/clothing/head/jobs.dm:38]: eh, sure, why not

worse case scenario, it's just a redundant check

Comment (ATH1909) [reply code/modules/clothing/head/jobs.dm:38]: eh, worst case scenario, we have a mouse puppeting a guy's unconscious body/sleeping corpse, right? then it'd be like that wallace and gromit movie with the trousers.

**Comment by Asha108** [comment]
Player controlled chef.

File: code/modules/clothing/head/jobs.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/clothing/head/jobs.dm:38]: dead guys too.

Comment (ArcaneDefence) [reply code/modules/clothing/head/jobs.dm:38]: Is this a means to break out of a pull while cuffed?

File: code/modules/clothing/head/jobs.dm (hunk 2 above)
Comment (ATH1909) [reply code/modules/clothing/head/jobs.dm:21]: do we use snake_case for paths? "chefhat" isn't in snake_case...

**Comment by ishitbyabullet** [comment]
Honestly, having it be something they can buy would probably be a neat thing. Could be a really expensive premium item or a traitor item.

I mean, really, what harm would this do if it wasn't admin-only? What - are sentient mice going to become the new xenobio strategy? That's just retarded.

**Comment by Doosteroni** [comment]
Those damn powergamers turning themselves into rats!

**Comment by Tharcoonvagh** [comment]
Make it a smite as well

File: code/modules/clothing/head/jobs.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/clothing/head/jobs.dm:38]: Yes.

File: code/modules/clothing/head/jobs.dm (hunk 2 above)
Comment (ShizCalev) [reply code/modules/clothing/head/jobs.dm:21]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#type-paths-must-be-snake-case

yes, and it's a new thing

**Comment by ATH1909** [comment]
> Honestly, having it be something they can buy would probably be a neat thing. Could be a really expensive premium item or a traitor item.
> 
> I mean, really, what harm would this do if it wasn't admin-only? What - are sentient mice going to become the new xenobio strategy? That's just stupid.

sirthisisawebedit

For real, though, I'd be down with someone adding it to the contraband section of one of the chef vending machines in a future PR.

**Comment by Asha108** [comment]
Why not include it in the blackmarket uplink?

**Comment by davethwave** [comment]
How do you actually use it though? When a mouse moves in it, it *uncurls* and gets out of the hat.

**Pull request merged**